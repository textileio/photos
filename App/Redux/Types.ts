import { StateType } from 'typesafe-actions'
import { PersistPartial } from 'redux-persist'

import RootReducer from './RootReducer'
import { AccountAction } from './AccountRedux'
import { AuthAction } from './AuthRedux'
import { CameraRollAction } from './CameraRollRedux'
import { ContactsAction } from './ContactsRedux'
import { PhotoViewingAction } from './PhotoViewingRedux'
import { PreferencesAction } from './PreferencesRedux'
import { NotificationsAction } from './NotificationsRedux'
import { ThreadsAction } from './ThreadsRedux'
import { TriggersAction } from './TriggersRedux'
import { UIAction } from './UIRedux'
import { UploadingImagesAction } from './UploadingImagesRedux'
import { StartupAction } from './StartupRedux'
import { MigrationAction } from './MigrationRedux'
import { TextileEventsActions } from './TextileEventsRedux'
import { GroupAction } from '../features/group'

export type RootState = StateType<typeof RootReducer> & PersistPartial
export type RootAction =
  AccountAction |
  AuthAction |
  CameraRollAction |
  ContactsAction |
  PhotoViewingAction |
  PreferencesAction |
  NotificationsAction |
  ThreadsAction |
  TriggersAction |
  UIAction |
  UploadingImagesAction |
  StartupAction |
  MigrationAction |
  TextileEventsActions |
  GroupAction
