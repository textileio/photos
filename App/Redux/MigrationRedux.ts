import { createAction, getType, ActionType } from 'typesafe-actions'
import { Protobufs } from '@textile/react-native-sdk'
import { ContactInfo } from '@textile/react-native-sdk'

const actions = {
  migrationNeeded: createAction('@migration/MIGRATION_NEEDED'),
  requestMigration: createAction('@migration/REQUEST_MIGRATION'),
  cancelMigration: createAction('@migration/CANCEL_MIGRATION'),
  retryMigration: createAction('@migration/RETRY_MIGRATION'),
  migrationStarted: createAction('@migration/MIGRATION_STARTED'),
  migrationComplete: createAction('@migration/MIGRATION_COMPLETE'),
  migrationError: createAction('@migration/MIGRATION_ERROR', (resolve) => (error: any) => resolve({ error })),
  requestRunRecurringMigrationTasks: createAction('@migration/REQUEST_RECURRING_MIGRATION'),
  peerAnnouncement: createAction(
    '@migration/PEER_ANNOUNCEMENT',
    (resolve) => (peerDetails: PeerDetails) => resolve({ peerDetails })
  ),
  photoMigration: createAction(
    '@migration/PHOTO_MIGRATION_SETUP',
    (resolve) => (photos: MigrationPhoto[]) => resolve({ photos })
  ),
  peerAnnouncementSuccess: createAction('@migration/PEER_ANNOUNCEMENT_SUCCESS'),
  connectToPeers: createAction(
    '@migration/CONNECT_TO_PEERS',
    (resolve) => (peers: string[]) => resolve({ peers })
  ),
  connectionSuccess: createAction(
    '@migration/CONNECTION_SUCCESS',
    (resolve) => (peer: string) => resolve({ peer })
  ),
  insertDownload: createAction(
    '@migration/INSERT_DOWNLOAD',
    (resolve) => (photoId: string, path: string) => resolve({ photoId, path })
  ),
  downloadStarted: createAction(
    '@migration/DOWNLOAD_STARTED',
    (resolve) => (photoId: string, statusCode: number, contentLength: number) => resolve({ photoId, statusCode, contentLength })
  ),
  downloadProgress: createAction(
    '@migration/DOWNLOAD_PROGRESS',
    (resolve) => (photoId: string, bytesWritten: number) => resolve({ photoId, bytesWritten })
  ),
  downloadComplete: createAction(
    '@migration/DOWNLOAD_COMPLETE',
    (resolve) => (photoId: string, statusCode: number, bytesWritten: number) => resolve({ photoId, statusCode, bytesWritten })
  ),
  downloadError: createAction(
    '@migration/DOWNLOAD_ERROR',
    (resolve) => (photoId: string, error: any) => resolve({ photoId, error })
  ),
  insertLocalProcessingTask: createAction(
    '@migration/INSERT_LOCAL_PROCESSING_TASK',
    (resolve) => (photoId: string) => resolve({ photoId })
  ),
  localProcessingTaskComplete: createAction(
    '@migration/INSERT_LOCAL_PROCESSING_TASK_COMPLETE',
    (resolve) => (photoId: string, preparedFiles: Protobufs.IMobilePreparedFiles) => resolve({ photoId, preparedFiles })
  ),
  localProcessingTaskError: createAction(
    '@migration/INSERT_LOCAL_PROCESSING_TASK_ERROR',
    (resolve) => (photoId: string, error: any) => resolve({ photoId, error })
  ),
  insertUpload: createAction(
    '@migration/INSERT_UPLOAD',
    (resolve) => (photoId: string, path: string, totalBytesExpectedToSend: number) => resolve({ photoId, path, totalBytesExpectedToSend })
  ),
  uploadStarted: createAction(
    '@migration/UPLOAD_STARTED',
    (resolve) => (photoId: string) => resolve({ photoId })
  ),
  uploadProgress: createAction(
    '@migration/UPLOAD_PROGRESS',
    (resolve) => (photoId: string, totalBytesExpectedToSend: number, totalBytesSent: number) => resolve({ photoId, totalBytesExpectedToSend, totalBytesSent })
  ),
  uploadComplete: createAction(
    '@migration/UPLOAD_COMPLETE',
    (resolve) => (photoId: string, statusCode: number) => resolve({ photoId, statusCode })
  ),
  uploadError: createAction(
    '@migration/UPLOAD_ERROR',
    (resolve) => (photoId: string, error: any) => resolve({ photoId, error })
  )
}

export type MigrationAction = ActionType<typeof actions>

export interface MigrationPhoto {
  id: string,
  key: string
}

export interface PhotoDownload {
  readonly photoId: string
  readonly path: string
  readonly contentLength?: number
  readonly bytesWritten: number
  readonly statusCode?: number
  readonly error?: string
  readonly status: 'pending' | 'downloading' | 'complete' | 'error'
}

export interface PhotoDownloads {
  readonly [key: string]: PhotoDownload
}

export interface LocalProcessingTask {
  readonly photoId: string
  readonly preparedFiles?: Protobufs.IMobilePreparedFiles
  readonly status: 'processing' | 'complete' | 'error'
  readonly error?: string
}

export interface LocalProcessingTasks {
  readonly [key: string]: LocalProcessingTask
}

export interface PhotoUpload {
  readonly photoId: string
  readonly path: string
  readonly statusCode?: number
  readonly totalBytesExpectedToSend: number
  readonly totalBytesSent: number
  readonly error?: string
  readonly status: 'pending' | 'uploading' | 'complete' | 'error'
}

export interface PhotoUploads {
  readonly [key: string]: PhotoUpload
}

export interface PeerDetails {
  readonly previousPeerId: string,
  readonly previousUsername?: string,
  readonly currentContactInfo?: ContactInfo
}

export interface MigrationState {
  readonly status: 'none' | 'pending' | 'processing' | 'complete' | 'cancelled'
  readonly peerAnnouncement?: {
    readonly peerDetails: PeerDetails,
    readonly status: 'pending' | 'complete'
  }
  readonly network?: ReadonlyArray<string>,
  readonly migrationPhotos?: ReadonlyArray<MigrationPhoto>
  readonly photoDownloads?: PhotoDownloads
  readonly localProcessingTasks?: LocalProcessingTasks
  readonly photoUploads?: PhotoUploads
  readonly error?: string
}

const initialState: MigrationState = {
  status: 'none'
}

export function reducer(state: MigrationState = initialState, action: MigrationAction): MigrationState {
  switch (action.type) {
    case getType(actions.migrationNeeded):
      return { ...state, status: 'pending' }
    case getType(actions.cancelMigration):
      return { ...state, status: 'cancelled' }
    case getType(actions.retryMigration):
      return { status: 'pending' }
    case getType(actions.migrationStarted):
      return { ...state, status: 'processing' }
    case getType(actions.migrationComplete):
      return { ...state, status: 'complete' }
    case getType(actions.migrationError): {
      const { error } = action.payload
      const message = error.message as string || error as string || 'unknown error'
      return { ...state, error: message }
    }
    case getType(actions.peerAnnouncement): {
      const { peerDetails } = action.payload
      return { ...state, peerAnnouncement: { peerDetails, status: 'pending' } }
    }
    case getType(actions.peerAnnouncementSuccess): {
      return { ...state, peerAnnouncement: { ...state.peerAnnouncement!, status: 'complete' } }
    }
    case getType(actions.connectToPeers): {
      const network = action.payload.peers
      return { ...state, network }
    }
    case getType(actions.connectionSuccess): {
      const { peer } = action.payload
      if (!state.network) {
        return state
      }
      const peers = state.network.filter((p) => p !== peer)
      return { ...state, network: peers }
    }
    case getType(actions.photoMigration): {
      const { photos } = action.payload
      return { ...state, migrationPhotos: photos }
    }
    case getType(actions.insertDownload): {
      const { photoId, path } = action.payload
      const photoDownloads: PhotoDownloads = { ...state.photoDownloads, [photoId]: { photoId, path, bytesWritten: 0, status: 'pending' }}
      return { ...state, photoDownloads }
    }
    case getType(actions.downloadStarted): {
      const { photoId, statusCode, contentLength } = action.payload
      const download = state.photoDownloads![photoId]
      const updatedDownload: PhotoDownload = { ...download, statusCode, contentLength, status: 'downloading' }
      const photoDownloads: PhotoDownloads = { ...state.photoDownloads, [photoId]: updatedDownload }
      return { ...state, photoDownloads }
    }
    case getType(actions.downloadProgress): {
      const { photoId, bytesWritten } = action.payload
      const download = state.photoDownloads![photoId]
      const updatedDownload: PhotoDownload = { ...download, bytesWritten }
      const photoDownloads: PhotoDownloads = { ...state.photoDownloads, [photoId]: updatedDownload }
      return { ...state, photoDownloads }
    }
    case getType(actions.downloadComplete): {
      const { photoId, statusCode, bytesWritten } = action.payload
      const download = state.photoDownloads![photoId]
      const updatedDownload: PhotoDownload = { ...download, statusCode, bytesWritten, status: 'complete' }
      const photoDownloads: PhotoDownloads = { ...state.photoDownloads, [photoId]: updatedDownload }
      return { ...state, photoDownloads }
    }
    case getType(actions.downloadError): {
      const { photoId, error } = action.payload
      const download = state.photoDownloads![photoId]
      const message = error.message as string || error as string || 'unknown error'
      const updatedDownload: PhotoDownload = { ...download, error: message, status: 'error' }
      const photoDownloads: PhotoDownloads = { ...state.photoDownloads, [photoId]: updatedDownload }
      return { ...state, photoDownloads }
    }
    case getType(actions.insertLocalProcessingTask): {
      const { photoId } =  action.payload
      return { ...state, localProcessingTasks: { ...state.localProcessingTasks, [photoId]: { photoId, status: 'processing' } } }
    }
    case getType(actions.localProcessingTaskComplete): {
      const { photoId, preparedFiles } = action.payload
      const taskData = state.localProcessingTasks![photoId]
      const updated: LocalProcessingTask = { ...taskData, status: 'complete', preparedFiles }
      return { ...state, localProcessingTasks: { ...state.localProcessingTasks, [photoId]: updated } }
    }
    case getType(actions.localProcessingTaskError): {
      const { photoId, error } = action.payload
      const taskData = state.localProcessingTasks![photoId]
      const message = error.message as string || error as string || 'unknown error'
      const updated: LocalProcessingTask = { ...taskData, status: 'error', error: message }
      return { ...state, localProcessingTasks: { ...state.localProcessingTasks, [photoId]: updated } }
    }
    case getType(actions.insertUpload): {
      const { photoId, path, totalBytesExpectedToSend } = action.payload
      const photoUploads: PhotoUploads = { ...state.photoUploads, [photoId]: { photoId, path, totalBytesExpectedToSend, totalBytesSent: 0, status: 'pending' }}
      return { ...state, photoUploads }
    }
    case getType(actions.uploadStarted): {
      const { photoId } = action.payload
      const upload = state.photoUploads![photoId]
      const updatedUpload: PhotoUpload = { ...upload, status: 'uploading' }
      const photoUploads: PhotoUploads = { ...state.photoUploads, [photoId]: updatedUpload }
      return { ...state, photoUploads }
    }
    case getType(actions.uploadProgress): {
      const { photoId, totalBytesExpectedToSend, totalBytesSent } = action.payload
      const upload = state.photoUploads![photoId]
      const updatedUpload: PhotoUpload = { ...upload, totalBytesExpectedToSend, totalBytesSent }
      const photoUploads: PhotoUploads = { ...state.photoUploads, [photoId]: updatedUpload }
      return { ...state, photoUploads }
    }
    case getType(actions.uploadComplete): {
      const { photoId, statusCode } = action.payload
      const upload = state.photoUploads![photoId]
      const updatedUpload: PhotoUpload = { ...upload, statusCode, status: 'complete' }
      const photoUploads: PhotoUploads = { ...state.photoUploads, [photoId]: updatedUpload }
      return { ...state, photoUploads }
    }
    case getType(actions.uploadError): {
      const { photoId, error } = action.payload
      const upload = state.photoUploads![photoId]
      const message = error.message as string || error as string || 'unknown error'
      const updatedUpload: PhotoUpload = { ...upload, error: message, status: 'error' }
      const photoUploads: PhotoUploads = { ...state.photoUploads, [photoId]: updatedUpload }
      return { ...state, photoUploads }
    }
    default:
      return state
  }
}

export default actions
