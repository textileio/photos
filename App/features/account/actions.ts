import { createAction, createAsyncAction } from 'typesafe-actions'
import { IContact } from '@textile/react-native-sdk'

import { SharedImage } from '../group/add-photo/models'

export const chooseProfilePhoto = createAsyncAction(
  'account/CHOOSE_PROFILE_PHOTO_REQUEST',
  'account/CHOOSE_PROFILE_PHOTO_SUCCESS',
  'account/CHOOSE_PROFILE_PHOTO_ERROR'
)<void, { image: SharedImage; data: string }, any>()

export const cancelProfilePhotoUpdate = createAction(
  'account/CANCEL_PROFILE_PHOTO_UPDATE',
  resolve => {
    return () => resolve()
  }
)

export const refreshProfileRequest = createAction(
  'account/REFRESH_PROFILE_REQUEST'
)

export const refreshProfileSuccess = createAction(
  'account/REFRESH_PROFILE_SUCCESS',
  resolve => {
    return (profile: IContact) => resolve({ profile })
  }
)

export const profileError = createAction('account/PROFILE_ERROR', resolve => {
  return (error: any) => resolve({ error })
})

export const refreshPeerIdRequest = createAction(
  'account/REFRESH_PEER_ID_REQUEST'
)

export const refreshPeerIdSuccess = createAction(
  'account/REFRESH_PEER_ID_SUCCESS',
  resolve => {
    return (peerId: string) => resolve({ peerId })
  }
)

export const refreshPeerIdError = createAction(
  'account/REFRESH_PEER_ID_ERROR',
  resolve => {
    return (error: any) => resolve({ error })
  }
)

export const refreshAddressRequest = createAction(
  'account/REFRESH_ADDRESS_REQUEST'
)

export const refreshAddressSuccess = createAction(
  'account/REFRESH_ADDRESS_SUCCESS',
  resolve => {
    return (address: string) => resolve({ address })
  }
)

export const refreshAddressError = createAction(
  'account/REFRESH_ADDRESS_ERROR',
  resolve => {
    return (error: any) => resolve({ error })
  }
)

export const setUsernameRequest = createAction(
  'account/SET_USERNAME_REQUEST',
  resolve => {
    return (username: string) => resolve({ username })
  }
)

export const setAvatar = createAsyncAction(
  'account/SET_AVATAR_REQUEST',
  'account/SET_AVATAR_SUCCESS',
  'account/SET_AVATAR_FAILURE'
)<SharedImage, void, any>()

export const setRecoveryPhrase = createAction(
  'account/SET_RECOVERY_PHRASE',
  resolve => {
    return (recoveryPhrase: string) => resolve({ recoveryPhrase })
  }
)
