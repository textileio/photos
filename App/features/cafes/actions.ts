import { createAsyncAction } from 'typesafe-actions'

// Registering with a new cafe requires the cafe URL and string
// With the new SDK, the URL needs to be changed to the peer id
export const registerCafe = createAsyncAction(
  'cafes/REGISTER_CAFE_REQUEST',
  'cafes/REGISTER_CAFE_SUCCESS',
  'cafes/REGISTER_CAFE_ERROR'
)<
  {
    peerId: string
    token: string
    success?: () => void
  },
  string,
  {
    peerId: string
    error: any
  }
>()

// Deregistering from a cafe requires the cafe's peer id
export const deregisterCafe = createAsyncAction(
  'cafes/DEREGISTER_CAFE_REQUEST',
  'cafes/DEREGISTER_CAFE_SUCCESS',
  'cafes/DEREGISTER_CAFE_ERROR'
)<
  {
    id: string
    success?: () => void
  },
  string,
  {
    id: string
    error: any
  }
>()
