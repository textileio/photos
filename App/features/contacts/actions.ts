import { createAction } from 'typesafe-actions'
import { pb } from '@textile/react-native-sdk'
import Contacts from 'react-native-contacts'

export const getContactsRequest = createAction('GET_CONTACTS_REQUEST')

export const getContactsSuccess = createAction('GET_CONTACT_SUCCESS', (resolve) => {
  return (contacts: ReadonlyArray<pb.IContact>) => resolve({ contacts })
})

export const searchRequest = createAction('@contacts/SEARCH_REQUEST', (resolve) => {
  return (searchString: string) => resolve({ searchString })
})

export const searchStarted = createAction('@contacts/SEARCH_STARTED')

export const searchResultTextile = createAction('@contacts/SEARCH_RESULT_TEXTILE', (resolve) => {
  return (result: pb.IContact) => resolve({ result })
})

export const textileSearchComplete = createAction('@contacts/TEXTILE_SEARCH_COMPLETE')

export const searchResultsAddressBook = createAction('@contacts/SEARCH_RESULTS_ADDRESS_BOOK', (resolve) => {
  return (results: ReadonlyArray<Contacts.Contact>) => resolve({ results })
})

export const searchErrorTextile = createAction('@contacts/SEARCH_ERROR_TEXTILE', (resolve) => {
  return (error: any) => resolve({ error })
})

export const searchErrorAddressBook = createAction('@contacts/SEARCH_ERROR_ADDRESS_BOOK', (resolve) => {
  return (error: any) => resolve({ error })
})

export const clearSearch = createAction('@contacts/CLEAR_SEARCH')

export const addContactRequest = createAction('@contacts/ADD_CONTACT_REQUEST', (resolve) => {
  return (contact: pb.IContact) => resolve({ contact })
})

export const addContactSuccess = createAction('@contacts/ADD_CONTACT_SUCCESS', (resolve) => {
  return (contact: pb.IContact) => resolve({ contact })
})

export const addContactError = createAction('@contacts/ADD_CONTACT_ERROR', (resolve) => {
  return (contact: pb.IContact, error: any) => resolve({ contact, error })
})

export const clearAddContact = createAction('@contacts/CLEAR_ADD_CONTACT_ERROR', (resolve) => {
  return (contact: pb.IContact) => resolve({ contact })
})

export const authorInviteRequest = createAction('@contacts/AUTHOR_INVITE_REQUEST', (resolve) => {
  return (contact: Contacts.Contact) => resolve({ contact })
})
