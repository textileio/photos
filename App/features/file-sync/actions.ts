import { createAction, createAsyncAction } from 'typesafe-actions'

export const begin = createAction('file-sync/BEGIN', resolve => {
  return (group: string, id: string) => resolve({ group, id })
})

export const uploadProgress = createAction(
  'file-sync/UPLOAD_PROGRESS',
  resolve => {
    return (id: string, progress: number) => resolve({ id, progress })
  }
)

export const uploadComplete = createAction(
  'file-sync/UPLOAD_COMPLETE',
  resolve => {
    return (id: string, responseCode: string, responseBody: string) =>
      resolve({ id, responseCode, responseBody })
  }
)

export const error = createAction('file-sync/ERROR', resolve => {
  return (id: string, error: string) => resolve({ id, error })
})
