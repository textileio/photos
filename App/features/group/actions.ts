import { feedActions as feed } from './feed'
import { addMessageActions as addMessage } from './add-message'
import { addPhotoActions as addPhoto } from './add-photo'

export { feed, addMessage, addPhoto }
