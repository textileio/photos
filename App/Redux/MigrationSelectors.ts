import { RootState } from './Types'
import { PeerDetails, MigrationPhoto, PhotoDownload, LocalProcessingTask } from './MigrationRedux'

export function overallUploadProgress(state: RootState) {
  let totalSize = 0
  let totalUploaded = 0
  if (state.migration.photoUploads) {
    for (const photoId in state.migration.photoUploads) {
      if (state.migration.photoUploads[photoId]) {
        const upload = state.migration.photoUploads[photoId]
        totalSize = totalSize + (upload.totalBytesExpectedToSend || 0)
        totalUploaded = totalUploaded + (upload.totalBytesSent || 0)
      }
    }
  }
  if (totalSize > 0) {
    return totalUploaded / totalSize
  } else {
    return undefined
  }
}

export function allLocalProcessingTasks(state: RootState) {
  return Object.keys(state.migration.localProcessingTasks || {})
    .map((key) => state.migration.localProcessingTasks![key] as LocalProcessingTask)
}

export function completeLocalProcessingTasks(state: RootState) {
  return Object.keys(state.migration.localProcessingTasks || {})
    .map((key) => state.migration.localProcessingTasks![key] as LocalProcessingTask)
    .filter((task) => task.status === 'complete')
}

export function completeLocalProcessingTasksCount(state: RootState) {
  return Object.keys(state.migration.localProcessingTasks || {})
    .map((key) => state.migration.localProcessingTasks![key] as LocalProcessingTask)
    .filter((task) => task.status === 'complete' || task.status === 'error')
    .length
}

export function localProcessingTasksCount(state: RootState) {
  return Object.keys(state.migration.localProcessingTasks || {}).length
}

export function completeDownloads(state: RootState) {
  return Object.keys(state.migration.photoDownloads || {})
    .map((key) => state.migration.photoDownloads![key] as PhotoDownload)
    .filter((download) => download.status === 'complete')
}

export function completeDownloadsCount(state: RootState) {
  return Object.keys(state.migration.photoDownloads || {})
    .map((key) => state.migration.photoDownloads![key] as PhotoDownload)
    .filter((download) => download.status === 'complete' || download.status === 'error')
    .length
}

export function downloadsCount(state: RootState) {
  return Object.keys(state.migration.photoDownloads || {}).length
}

export function getAnnouncement(state: RootState) {
  return state.migration.peerAnnouncement
}

export function getNetwork(state: RootState): ReadonlyArray<string> {
  return state.migration.network || []
}

export function getMigrationPhotos(state: RootState): ReadonlyArray<MigrationPhoto> {
  return state.migration.migrationPhotos || []
}
