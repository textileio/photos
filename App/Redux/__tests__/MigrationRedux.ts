import actions, { reducer } from '../MigrationRedux'
import { getAnnouncement, getNetwork } from '../MigrationSelectors'
import { PeerDetails, MigrationState } from '../MigrationRedux'
import { RootState } from '../Types'

const initialState = reducer(undefined, {} as any)
const profile: MigrationState = {
  photoDownloads: {},
  localProcessingTasks: {},
  status: 'none'
}

const peerDetails: PeerDetails = {
  previousPeerId: 'XYZ-UVW-RST',
  previousUsername: 'jest test',
  currentContactInfo: {
    id: 'id',
    address: 'address',
    created: 'created',
    updated: 'updated'
  }
}

const network: string[] = [
  'AAA',
  'BBB',
  'CCC'
]

describe('migration', () => {
  describe('initial state', () => {
    it('should match snapshot', () => {
      expect(initialState).toMatchSnapshot()
    })
  })
  describe('migration', () => {
    let migrationState = initialState
    it('should store peer details to migration', () => {
      migrationState =  reducer(
        migrationState,
        actions.peerAnnouncement(peerDetails)
      )
      expect(migrationState.peerAnnouncement).toBeDefined()
      expect(migrationState.peerAnnouncement!.peerDetails).toEqual(peerDetails)
    })
    it('should select correct announcement details', () => {
      const selected = getAnnouncement({migration: migrationState} as RootState)
      expect(selected).toBeDefined()
      expect(selected!.peerDetails).toEqual(peerDetails)
    })
    it('should store network details for migration', () => {
      migrationState =  reducer(
        migrationState,
          actions.connectToPeers(
            network
          )
        )
      expect(migrationState.network).toEqual(network)
    })
    it('should select correct network details', () => {
      const selected = getNetwork({migration: migrationState} as RootState)
      expect(selected).toEqual(network)
    })
  })
})
