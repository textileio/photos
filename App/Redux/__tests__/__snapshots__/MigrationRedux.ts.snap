// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`migration initial state should match snapshot 1`] = `
Object {
  "status": "none",
}
`;
